media/system/js/treeselectmenu.js

```diff
@@ -2,117 +2,136 @@
  * @copyright  Copyright (C) 2005 - 2017 Open Source Matters, Inc. All rights reserved.
  * @license    GNU General Public License version 2 or later; see LICENSE.txt
  */
-jQuery(function($)
+Joomla = window.Joomla || {};
+
+(function(Joomla)
 {
-	var treeselectmenu = $('div#treeselectmenu').html();
+	var treeselectmenu = document.getElementById("treselectmenu");
+	var treeselect = document.getElementsByClassName("treeselect");
+	var element = treeselect.getElementsByTagName("li");
 
-	$('.treeselect li').each(function()
+	element.forEach(function()
 	{
-		$li = $(this);
-		$div = $li.find('div.treeselect-item:first');
+		var li = event.target;
+		var div = li.querySelector('div.treeselect-item:first');
+		var span = document.createElement("span");
 
 		// Add icons
-		$li.prepend('<span class="icon-"></span>');
+		li = document.createElement('span');
+		li.className = "icon-";
 
-		if ($li.find('ul.treeselect-sub').length) {
+		if (li.querySelector('ul.treeselect-sub').length) {
 			// Add classes to Expand/Collapse icons
-			$li.find('span.icon-').addClass('treeselect-toggle fa-chevron-down');
+			li.querySelectorAll('span.icon-').classList.add('treeselect-toggle fa-chevron-down');
 
 			// Append drop down menu in nodes
-			$div.find('label:first').after(treeselectmenu);
+			var element = div.querySelector('label:first');
+			element.parentNode.insertBefore(treeselectmenu, element);
 
-			if (!$li.find('ul.treeselect-sub ul.treeselect-sub').length) {
-				$li.find('div.treeselect-menu-expand').remove();
+			if (!li.querySelector('ul.treeselect-sub ul.treeselect-sub').length) {
+				li.querySelector('div.treeselect-menu-expand').classList.remove();
```

Comment:
refer https://api.jquery.com/remove/
`.remove()` removes the element. Not the classses.